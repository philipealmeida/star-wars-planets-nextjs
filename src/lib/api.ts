const SWAPI_BASE_URL = 'https://swapi.dev/api';

export const fetchPlanets = async (page: number = 1, search?: string) => {
  const searchParams = new URLSearchParams({
    page: page.toString(),
    ...(search && { search }),
  });
  
  const response = await fetch(
    `${SWAPI_BASE_URL}/planets/?${searchParams.toString()}`
  );
  
  if (!response.ok) {
    throw new Error('Failed to fetch planets');
  }
  
  return response.json();
};

export const fetchPlanetById = async (id: string) => {
  const response = await fetch(`${SWAPI_BASE_URL}/planets/${id}`);
  
  if (!response.ok) {
    throw new Error('Failed to fetch planet details');
  }
  
  return response.json();
};