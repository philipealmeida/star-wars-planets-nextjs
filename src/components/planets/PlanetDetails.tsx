import Link from 'next/link';
import { Planet } from '@/lib/types';
import styles from '@/styles/components/planets/PlanetDetails.module.css';

interface PlanetDetailsProps {
  planet: Planet;
}

export default function PlanetDetails({ planet }: PlanetDetailsProps) {
  return (
    <div className={styles.details}>
      <h1 className={styles.planetName}>{planet.name}</h1>
      <div className={styles.content}>
        <InfoItem label="Climate" value={planet.climate} />
        <InfoItem label="Terrain" value={planet.terrain} />
        <InfoItem label="Population" value={planet.population} />
        <InfoItem label="Diameter" value={`${planet.diameter} km`} />
        <InfoItem label="Gravity" value={planet.gravity} />
        <InfoItem label="Surface Water" value={`${planet.surface_water}%`} />
      </div>
      <Link href="/planets" className={styles.back}>
        ‚Üê Back to Planets
      </Link>
    </div>
  );
}

function InfoItem({ label, value }: { label: string; value: string }) {
  return (
    <div className={styles.infoItem}>
      <span className={styles.label}>{label}</span>
      <span className={styles.value}>{value}</span>
    </div>
  );
}