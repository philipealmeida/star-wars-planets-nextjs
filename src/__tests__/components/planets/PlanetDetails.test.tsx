import { Planet } from '@/lib/types';
import { render, screen } from '@testing-library/react';
import PlanetDetails from '@/components/planets/PlanetDetails';

const mockPlanet: Planet = {
    name: 'Tatooine',
    climate: 'Arid',
    terrain: 'Desert',
    population: '200000',
    diameter: '10465',
    gravity: '1 standard',
    surface_water: '1',
    url: 'https://swapi.dev/api/planets/1/',
    rotation_period: '',
    orbital_period: '',
    residents: [],
    films: [],
    created: '',
    edited: ''
};

const renderComponent = (props: { planet: Planet }) => {
  return render(<PlanetDetails {...props} />);
};

describe('PlanetDetails', () => {
  it('matches snapshot', () => {
    const { container } = renderComponent({ planet: mockPlanet });
    expect(container).toMatchSnapshot();
  });

  it('displays all planet details correctly', () => {
    renderComponent({ planet: mockPlanet });
    
    expect(screen.getByText('Tatooine')).toBeInTheDocument();
    expect(screen.getByText('Arid')).toBeInTheDocument();
    expect(screen.getByText('Desert')).toBeInTheDocument();
    expect(screen.getByText('200000')).toBeInTheDocument();
    expect(screen.getByText('10465 km')).toBeInTheDocument();
    expect(screen.getByText('1 standard')).toBeInTheDocument();
    expect(screen.getByText('1%')).toBeInTheDocument();
  });

  it('renders back link correctly', () => {
    renderComponent({ planet: mockPlanet });
    const backLink = screen.getByText('← Back to Planets');
    expect(backLink).toHaveAttribute('href', '/planets');
  });
});