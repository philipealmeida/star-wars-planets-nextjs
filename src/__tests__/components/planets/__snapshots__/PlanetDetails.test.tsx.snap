// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanetDetails matches snapshot 1`] = `
<div>
  <div
    class="details"
  >
    <h1
      class="planetName"
    >
      Tatooine
    </h1>
    <div
      class="content"
    >
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Climate
        </span>
        <span
          class="value"
        >
          Arid
        </span>
      </div>
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Terrain
        </span>
        <span
          class="value"
        >
          Desert
        </span>
      </div>
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Population
        </span>
        <span
          class="value"
        >
          200000
        </span>
      </div>
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Diameter
        </span>
        <span
          class="value"
        >
          10465 km
        </span>
      </div>
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Gravity
        </span>
        <span
          class="value"
        >
          1 standard
        </span>
      </div>
      <div
        class="infoItem"
      >
        <span
          class="label"
        >
          Surface Water
        </span>
        <span
          class="value"
        >
          1%
        </span>
      </div>
    </div>
    <a
      class="back"
      href="/planets"
    >
      ‚Üê Back to Planets
    </a>
  </div>
</div>
`;
